src/libexpr/primops/context.cc

```diff
@@ -37,8 +37,15 @@ static void prim_unsafeDiscardOutputDependency(EvalState & state, const PosIdx p
     auto s = state.coerceToString(pos, *args[0], context, "while evaluating the argument passed to builtins.unsafeDiscardOutputDependency");
 
     PathSet context2;
-    for (auto & p : context)
-        context2.insert(p.at(0) == '=' ? std::string(p, 1) : p);
+    for (auto && p : context) {
```

Comment:
We're concerned about the safety of `std::move(p)` a couple of lines down. `context` is local to this scope and we only consume it here, so it would be nice to avoid copying.

Reply:
At this point it could be a `&` or `&&`, does not really matter, as we need the `move` anyway. But it's a nice declaration of what we want to do with it.